packages-backend/express/src/auth.ts

```diff
@@ -104,15 +104,15 @@ const getConfiguredDefaultIssuer = (options: TSessionMiddlewareOptions) => {
 // Construct the audience from the given option + the request path.
 // If the request path is `/`, do not append it to the audience, otherwise
 // the token validation might fail because of mismatching audiences.
-const getConfiguredAudience = (
+export const getConfiguredAudience = (
   options: TSessionMiddlewareOptions,
   requestPath: string
 ) => {
   const url = new URL(options.audience);
   if (requestPath !== '/') {
-    url.pathname = requestPath;
+    return `${url.origin}${requestPath}`;
   }
-  return url.toString();
+  return url.origin;
```

Comment:
We have to change this now to omit the search params. I suggest to parse everything into the `URL` constructor and then return `origin` and `pathname`.

```js
const url = new URL(`${options.audience.replace(/\/?$/, '')}${requestPath}`);
if (requestPath === '/') {
  return url.origin;
}
return `${url.origin}${url.pathname}`;
```

Reply:
👍

Reply:
https://github.com/commercetools/merchant-center-application-kit/pull/2118/commits/2758c9d4bce4e1a7eeef36be517abd69c683d730